src/Xamarin.Android.Build.Tasks/Xamarin.Android.D8.targets

```diff
@@ -0,0 +1,137 @@
+        PrintSeedsOutput="$(IntermediateOutputPath)proguard\seeds.txt"
+        PrintUsageOutput="$(IntermediateOutputPath)proguard\usage.txt"
+        PrintMappingOutput="$(IntermediateOutputPath)proguard\mapping.txt"
+        ProguardInputJarFilter="$(_AndroidProguardInputJarFilter)"
+    />
+
+    <CreateItem 
+          Include="$(IntermediateOutputPath)proguard\__proguard_output__.jar"
+          Condition=" '$(AndroidLinkTool)' == 'proguard' And '$(_ProguardProjectConfiguration)' != '' ">
+      <Output TaskParameter="Include" ItemName="_AlternativeJarForAppD8" />
+    </CreateItem>
+
+    <R8
+        Condition=" '$(_UseR8)' == 'True' "
+        ToolPath="$(JavaToolPath)"
+        JavaMaximumHeapSize="$(JavaMaximumHeapSize)"
+        JavaOptions="$(JavaOptions)"
+        JarPath="$(R8JarPath)"
+        AndroidManifestFile="$(IntermediateOutputPath)android\AndroidManifest.xml"
+        OutputDirectory="$(IntermediateOutputPath)android\bin"
+        Debug="$(AndroidIncludeDebugSymbols)"
+        JavaPlatformJarPath="$(JavaPlatformJarPath)"
+        ClassesZip="$(IntermediateOutputPath)android\bin\classes.zip"
+        JavaLibrariesToEmbed="@(_JavaLibrariesToCompile);@(_InstantRunJavaReference)"
+        JavaLibrariesToReference="@(AndroidExternalJavaLibrary)"
+        EnableDesugar="$(AndroidEnableDesugar)"
+        AndroidSdkDirectory="$(_AndroidSdkDirectory)"
+        AndroidSdkBuildToolsPath="$(AndroidSdkBuildToolsPath)"
+        AcwMapFile="$(_AcwMapFile)"
+        ProguardCommonXamarinConfiguration="$(IntermediateOutputPath)proguard\proguard_xamarin.cfg"
+        ProguardGeneratedReferenceConfiguration="$(_ProguardProjectConfiguration)"
+        ProguardGeneratedApplicationConfiguration="$(IntermediateOutputPath)proguard\proguard_project_primary.cfg"
+        ProguardConfigurationFiles="$(ProguardConfigFiles)"
+        EnableShrinking="$(_R8EnableShrinking)"
+        EnableMultiDex="$(AndroidEnableMultiDex)"
+        MultiDexMainDexListFile="$(_AndroidMainDexListFile)"
+        CustomMainDexListFiles="@(MultiDexMainDexList)"
+        ExtraArguments="$(R8ExtraArguments)"
+    />
+    <D8
+        Condition=" '$(_UseR8)' != 'True' "
+        ToolPath="$(JavaToolPath)"
+        JavaMaximumHeapSize="$(JavaMaximumHeapSize)"
+        JavaOptions="$(JavaOptions)"
+        JarPath="$(D8JarPath)"
+        AndroidManifestFile="$(IntermediateOutputPath)android\AndroidManifest.xml"
+        OutputDirectory="$(IntermediateOutputPath)android\bin"
+        Debug="$(AndroidIncludeDebugSymbols)"
+        JavaPlatformJarPath="$(JavaPlatformJarPath)"
+        ClassesZip="$(IntermediateOutputPath)android\bin\classes.zip"
+        JavaLibrariesToEmbed="@(_JavaLibrariesToCompile);@(_InstantRunJavaReference)"
+        AlternativeJarLibrariesToEmbed="@(_AlternativeJarForAppD8)"
+        JavaLibrariesToReference="@(AndroidExternalJavaLibrary)"
+        EnableDesugar="$(AndroidEnableDesugar)"
+        ExtraArguments="$(D8ExtraArguments)"
+    />
```

Comment:
TODO after merged:
- The private Xamarin.Android sources use `<CompileToDalvik/>` for `Fast Deployment`, will need to optionally run `<D8/>` if `$(AndroidDexTool)='d8'`